### PR TITLE
Add other/tips section - part 1

### PR DESCRIPTION
**Pull request by kingthorin**
- Add initial tips and supporting images.

### CODE REVIEW

**Comment by kingthorin** [comment]
This is what I came up with for stand alone code to add a rule:

```js
// This script adds a replacer rule

var Control = Java.type("org.parosproxy.paros.control.Control");
var extReplacer = Control.getSingleton().getExtensionLoader().getExtension("ExtensionReplacer");

var replacerRule = Java.type("org.zaproxy.zap.extension.replacer.ReplacerParamRule");
var replacerParam = Java.type("org.zaproxy.zap.extension.replacer.ReplacerParam");
// (REQ_HEADER, REQ_HEADER_STR, REQ_BODY_STR, RESP_HEADER, RESP_HEADER_STR, RESP_BODY_STR)
var matchType = Java.type("org.zaproxy.zap.extension.replacer.ReplacerParamRule.MatchType");

newRule = new replacerRule("Change disable to enable", "", matchType.RESP_BODY_STR, "hidden", false, "hizzen", null, true, false);
extReplacer.getParams().addRule(newRule);
```

```java
//     public ReplacerParamRule(
//            String description,
//            String url,
//            MatchType matchType,
//            String matchString,
//            boolean matchRegex,
//            String replacement,
//            List<Integer> initiators,
//            boolean enabled,
//            boolean tokenProcessingEnabled)
```

I did discover that replacer doesn't prevent creation of duplicates. Not sure if we should care about that or not?

**Comment by kingthorin** [comment]
As discussed via IRC the code refs could be permalinked to the github repo with commit sha.

File: other/tips/replacer/match-and-replace/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+![](images/false-true-email.png)
+
+## Bypass WAF
+
+Bypassing WAF by adding some request headers.
+
+- Adding `X-Forwarded-Host: 127.0.0.1`
+
+![](images/bypass-waf.png)
```
Comment (kingthorin) [inline other/tips/replacer/match-and-replace/README.md:52]: Maybe this one should be header add if not present?

**Comment by kingthorin** [comment]
I think this is good to go. I'll add the script bits in another PR.

**Review by thc202 — APPROVED** [review]

**Comment by kingthorin** [comment]
For whoever adds the second review. I know the folder structure probably needs some intro'ish README.md files. I'll tackle that in the next (or a future) PR.

**Pull request merged**